zooniverse/front-end-monorepo
# Add translation dictionaries converted from PFE

## Package
app-project
lib-classifier
lib-react-components

## Linked Issue and/or Talk Post
[FEM Translations Project Board](https://github.com/zooniverse/front-end-monorepo/projects/15)

## Describe your changes

Custom node app used to convert json files: https://github.com/zooniverse/pfe-fem-translation-keys

This PR looks like a lot of file changes, but in reality the node app linked above does all the work, and these changes are simply copy-paste of dictionary files other than English.

Notes on PFE Dictionaries:
-  Although the English FEM dictionaries matches about 30% of PFE’s dictionary, languages other than English in fact have much less coverage.
- For some languages, their PFE dictionary has half English half other language. In those cases, I did not transfer a mix of languages to FEM dictionaries. For example, I removed any English phrases from the Arabic FEM dictionaries because when an FEM key is missing a translation, it will always default to English.

Other Changes
- I removed some locale options (el, kn, uk, and ko) because their PFE translation dictionary is purely English or they don't actually have a PFE translations dictionary. They just happened to be in PFE's [languageMenu.js](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/constants/languageMenu.js) when LocaleSwitcher was built in FEM.

Further Work
- I think further work on pluralization may be needed once these dictionary files are merged. Ideally, a native speaker of each language would translate FEM keys, and then do a spot check on the live website for incorrect pluralization.

## How to Review
- Make sure to `yarn bootstrap`
- To use LocaleSwitcher, add `import localeMenu from '@helpers/localeMenu'` to ProjectHeader, and reassign `availableLocales` to be `Object.keys(localeMenu)`. This way, every FEM-supported locale will be an option in the dropdown.
- You can also manually insert each locale abbreviation into the url.
- Visit a project that uses Survey Task because that particular page will pull all 5 translation dictionaries of the currently selected locale. Examples:
    - [https://local.zooniverse.org:3000/projects/wildcam/wildcam-darien/classify/workflow/3033?env=production](https://local.zooniverse.org:3000/projects/wildcam/wildcam-darien/classify/workflow/3033?env=production)
    - [https://local.zooniverse.org:3000/projects/markb-panoptes/survey-task-test-project/classify/workflow/3611?env=staging](https://local.zooniverse.org:3000/projects/markb-panoptes/survey-task-test-project/classify/workflow/3611?env=staging)
- As long as the page does not crash when selecting each locale, we’re good to go for now.
- Reminder that you can test which strings are considered static translations on any project page by switching to Test Language

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
